libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2033,3 +2063,48 @@ string YulUtilFunctions::revertReasonIfDebug(string const& _message)
 {
 	return revertReasonIfDebug(m_revertStrings, _message);
 }
+
+std::string YulUtilFunctions::tryDecodeErrorMessageFunction()
+{
+	string const functionName = "try_decode_error_message";
+
+	return m_functionCollector->createFunction(functionName, [&]() {
+		string const errorHash = FixedHash<4>(util::keccak256("Error(string)")).hex();
+		return util::Whiskers(R"(
+			function <functionName>() -> data {
+				data := mload(0x40)
```

Comment:
This is code copied from ContractCompiler right? Can you also use this function there?

Reply:
I should probably have done it for the calldata arrays already... it shouldn't be too hard, though - most of the required infrastructure is already there for the ``ABIFunction``s...

Reply:
The main "issue" is to not have them changed during optimization, so we'd need something like ``ABIFunctions::m_externallyUsedFunctions``...